src/branch-manager-service/shared/github.ts

```diff
@@ -86,13 +87,43 @@ export type GithubPullRequestEvent =
   PullRequestEventSynchronize |
   PullRequestEventUnlabeled;
 
-
 /** Pushes a status to github for a commmit. */
 export async function setStatusOnGithub(
     owner: string, repo: string, sha: string, state: 'error' | 'failure' | 'pending' | 'success',
     description: string = '', target_url: string = '', context: string = 'branch-manager') {
+  await authenticateGithubForInstallation(owner, repo);
   const statusParams: GithubApi.ReposCreateStatusParams = {
     context, description, owner, repo, sha, state, target_url 
-  }
+  };
   return github.repos.createStatus(statusParams);
 }
+
+/** Set github authentication to the the JWT for the github application. */
+export function authenticateGithubAsApplication() {
+  github.authenticate({type: 'app', token: generateJWT()});
+}
+
+/** Set github authentication to the token for the installation for a specific repo. */
+export async function authenticateGithubForInstallation(owner: string, repo: string) {
+  try {
+    authenticateGithubAsApplication();
+    const installationId = await github.apps.findRepoInstallation({owner, repo}).then(response => response.data.id);
+    const token = await github.apps.createInstallationToken({installation_id: installationId}).then(response => response.data.token);
+    github.authenticate({type: 'token', token});
+  } catch (e) {
+    console.error(e);
```

Comment:
Do we really just want to catch the error and print it _here_? Doesn't it continue that way and tries setting the status (above for example)?